test/gtest/plugins/transfer_handler.h

```diff
@@ -0,0 +1,346 @@
+#define __TRANSFER_HANDLER_H
+
+#include <absl/time/clock.h>
+#include <absl/time/time.h>
+#include "backend_engine.h"
+#include "common/nixl_log.h"
+#include "gtest/gtest.h"
+#include "memory_handler.h"
+
+template<nixl_mem_t srcMemType, nixl_mem_t dstMemType> class transferHandler {
+public:
+    transferHandler(std::shared_ptr<nixlBackendEngine> src_engine,
+                    std::shared_ptr<nixlBackendEngine> dst_engine,
+                    bool split_buf,
+                    int num_bufs)
+        : srcBackendEngine_(src_engine),
+          srcAgentName_(LOCAL_AGENT_NAME),
+          srcDevId_(0) {
+
+        bool remote_xfer = src_engine != dst_engine;
+        if (remote_xfer) {
+            CHECK(src_engine->supportsRemote()) << "Local engine does not support remote transfers";
+            dstBackendEngine_ = dst_engine;
+            dstAgentName_ = REMOTE_AGENT_NAME;
+            dstDevId_ = 1;
+            EXPECT_EQ(verifyConnInfo(), NIXL_SUCCESS);
+        } else {
+            CHECK(src_engine->supportsLocal()) << "Local engine does not support local transfers";
+            dstBackendEngine_ = src_engine;
+            dstAgentName_ = LOCAL_AGENT_NAME;
+            dstDevId_ = srcDevId_;
+        }
+
+        for (int i = 0; i < num_bufs; i++) {
+            srcMem_.emplace_back(
+                std::make_unique<memoryHandler<srcMemType>>(BUF_SIZE, srcDevId_ + i));
+            dstMem_.emplace_back(
+                std::make_unique<memoryHandler<dstMemType>>(BUF_SIZE, dstDevId_ + i));
+        }
+
+        if (dstBackendEngine_->supportsNotif()) setupNotifs("Test");
+
+        EXPECT_EQ(registerMems(), NIXL_SUCCESS);
+        EXPECT_EQ(prepMems(split_buf, remote_xfer), NIXL_SUCCESS);
+    }
+
+    ~transferHandler() {
+        EXPECT_EQ(srcBackendEngine_->unloadMD(xferLoadedMd_), NIXL_SUCCESS);
+        EXPECT_EQ(srcBackendEngine_->disconnect(dstAgentName_), NIXL_SUCCESS);
+        EXPECT_EQ(deregisterMems(), NIXL_SUCCESS);
+    }
+
+    void
+    testTransfer(nixl_xfer_op_t op) {
+        EXPECT_EQ(performTransfer(op), NIXL_SUCCESS);
+        EXPECT_EQ(verifyTransfer(op), NIXL_SUCCESS);
+    }
+
+    void
+    setLocalMem() {
+        for (size_t i = 0; i < srcMem_.size(); i++)
+            srcMem_[i]->setIncreasing(LOCAL_BUF_BYTE + i);
+    }
+
+    void
+    resetLocalMem() {
+        for (const auto &mem : srcMem_)
+            mem->reset();
+    }
+
+    void
+    checkLocalMem() {
+        for (size_t i = 0; i < srcMem_.size(); i++)
+            EXPECT_TRUE(srcMem_[i]->checkIncreasing(LOCAL_BUF_BYTE + i));
+    }
+
+private:
+    static constexpr uint8_t LOCAL_BUF_BYTE = 0x11;
+    static constexpr uint8_t XFER_BUF_BYTE = 0x22;
+    static constexpr size_t NUM_ENTRIES = 4;
+    static constexpr size_t ENTRY_SIZE = 16;
+    static constexpr size_t BUF_SIZE = NUM_ENTRIES * ENTRY_SIZE;
+    static constexpr std::string_view LOCAL_AGENT_NAME = "Agent1";
+    static constexpr std::string_view REMOTE_AGENT_NAME = "Agent2";
```

Comment:
These are now hardcoded both here in the infrastructure code and also in tests code. Consider making agent names constructor parameters for this class.